ml_tools/eolearn/tests/test_train_split.py

```diff
@@ -36,58 +36,65 @@ def test_bad_args(bad_arg: Any, bad_kwargs: Any) -> None:
         TrainTestSplitTask(INPUT_FEATURE, OUTPUT_FEATURE, bad_arg, **bad_kwargs)
 
 
-def test_train_split() -> None:
-    shape = (1000, 1000, 3)
-    data = np.random.randint(10, size=shape, dtype=int)
+@pytest.fixture(name="eopatch1")
+def eopatch1_fixture(seed: Optional[int] = None) -> EOPatch:
+    eopatch = EOPatch()
+    rng = np.random.default_rng(seed)
+    eopatch[INPUT_FEATURE] = rng.integers(0, 10, size=(1000, 1000, 3), dtype=int)
 
+    return eopatch
+
+
+@pytest.fixture(name="eopatch2")
+def eopatch2_fixture(seed: Optional[int] = None) -> EOPatch:
+    eopatch = EOPatch()
+    rng = np.random.default_rng(seed)
+    eopatch[INPUT_FEATURE] = rng.integers(0, 10, size=(1000, 1000, 3), dtype=int)
+
+    return eopatch
+
+
+def test_train_split(eopatch1: EOPatch) -> None:
+    """test hardcode some values in input feature and checks if hardcode values are split together"""
     indices = [(0, 2, 0, 2), (0, 2, 2, 4), (2, 4, 0, 2), (2, 4, 2, 4), (0, 4, 4, 8), (4, 8, 0, 4), (4, 8, 4, 8)]
     for index, (i_1, i_2, j_1, j_2) in enumerate(indices, 1):
-        data[i_1:i_2, j_1:j_2, :] = index * 11
-
-    patch = EOPatch()
-    patch[INPUT_FEATURE] = data
+        eopatch1[INPUT_FEATURE][i_1:i_2, j_1:j_2, :] = index * 11
```

Comment:
could be a problem if the scope of the fixture changes, because it would modify all of them

perhaps setting a scope explicitly + comment could be good enough

Reply:
done